orchestrator/modules/operators/console_output.py

```diff
@@ -0,0 +1,376 @@
+
+        Raises:
+            ValueError: If message is not an allowed type.
+        """
+        # Only accept typed messages
+        if not isinstance(
+            message, (RichConsoleSpinnerMessage, RichConsoleProgressMessage)
+        ):
+            raise ValueError("Unsupported message type for RichConsoleQueue.put")
+        with self._sempahore:
+            self._queue.append(message)
+
+    def get(
+        self, timeout=None
+    ) -> RichConsoleSpinnerMessage | RichConsoleProgressMessage | None:
+        """
+        Pop and return the next queued RichConsoleMessage, or None if queue is empty.
+
+        Args:
+            timeout (float|None): If None, block until a message is available. 0 for non-blocking, >0 for timed wait (currently ignored).
+
+        Returns:
+            RichConsoleSpinnerMessage | RichConsoleProgressMessage | None: The next message, or None if empty.
+        """
+        try:
+            with self._sempahore:
+                msg = self._queue.pop(0)
+        except IndexError:
+            msg = None
+        return msg
+
+
+def rich_console_spinner_from_message(msg: RichConsoleSpinnerMessage) -> Progress:
+    """
+    Construct a Rich Spinner instance configured from the given spinner message.
+
+    Args:
+        msg (RichConsoleSpinnerMessage): Spinner message to create Spinner from.
+    Returns:
+        Spinner: Instance of rich Spinner for use in Live UI updates.
+    """
+
+    bar = Progress(
+        SpinnerColumn(),
+        TextColumn("{task.description}"),
+        TimeElapsedColumn(),
+    )
+    bar.add_task(msg.label, total=None)
+    return bar
+
+
+def rich_console_progress_bar_from_message(msg: RichConsoleProgressMessage) -> Progress:
+    """
+    Construct a Rich Progress bar object from a given progress message.
+
+    Args:
+        msg (RichConsoleProgressMessage): Message with progress percent and label.
+    Returns:
+        Progress: Rich Progress bar object for dynamic UI updates.
+    """
+    bar = Progress(
+        SpinnerColumn(),
+        TextColumn("{task.description}"),
+        BarColumn(),
+        TextColumn("{task.percentage:>3.0f}%"),
+        TimeElapsedColumn(),
+    )
+    bar.add_task(msg.label, total=100, completed=msg.progress)
+    return bar
+
+
+def rich_console_progress_bar_update_from_message(
+    bar: Progress, msg: RichConsoleProgressMessage
+) -> bool:
+    """
+    Update an existing Rich Progress bar with the given progress message state.
+    Safely adds a new task if needed.
+
+    Args:
+        bar (Progress): Progress bar object to update.
+        msg (RichConsoleProgressMessage): Incoming message with updated percent.
+    Returns:
+        bool: True if progress is complete (progress >= 100), False otherwise.
+    """
+    # Assumes single task, always 0th index
+    if len(bar.task_ids) == 0:
+        task_id = bar.add_task(msg.label, total=100, completed=msg.progress)
+    else:
+        task_id = next(iter(bar.task_ids))
```

Comment:
This should also be true by construction with the way rich_console_progress_bar_from_message works, right?